### PR TITLE
Remove istio.io/for-service-account

### LINKED ISSUE #49915: Add option in istioctl to name waypoint
**Describe the feature request**
Came from a comment on another istioctl issue regarding waypoint updates (https://github.com/istio/istio/issues/49437#issuecomment-1992116273).

Task is to update istioctl to allow for users to name their waypoints.

**Describe alternatives you've considered**

Considered adding a new command like `nameWaypointCmd` but after talking with others, a `--name` flag seems to make more sense and is much simpler.

**Affected product area (please put an X in all that apply)**

[X] Ambient
[ ] Docs
[ ] Dual Stack
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

### LINKED ISSUE #50173: Remove `istio.io/for-service-account`
(This is used to request new product features, please visit <https://github.com/istio/istio/discussions> for questions on using Istio)

**Describe the feature request**

With #49700, WaypointServiceAccount "istio.io/for-service-account" has been useless, because the annotation is not checked anymore.

Is there any future use, otherwise please remove it

@ilrudie 

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[ ] Dual Stack
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

### PR DESCRIPTION
**Pull request by jmcshane**
**Please provide a description of this PR:**

Closes #50173 - removes unused WaypointServiceAccount concept
Closes #49915 - adds waypoint name to istioctl

With https://github.com/istio/istio/pull/49700, WaypointServiceAccount "istio.io/for-service-account" has been useless, because the annotation is not checked anymore.

### CODE REVIEW

**Comment by ilrudie** [comment]
Do we actually want to remove the flag from istioctl? It seems like although serviceAccount is not tied into binding the waypoint to a wl any more it is relevant for deploying the waypoint to the service account you want. Without this flag we're probably stuck with every waypoint using the exact same service account which will limit your L4 policy options wrt which waypoints are allowed to send traffic to which workloads.

**Comment by ilrudie** [comment]
FWIW, asserting l4 policy on inbound from a waypoint is still just planned, discussed but not implemented

**Comment by howardjohn** [comment]
I think we should have a 'name' for the waypoint. The SA is derived from
the name (following existing logic for ingress).

I agree we don't want to hardcode a "namespace" name. Also probably the
default should be something else we can bikeshed on. Something like
"default" or "default-waypoint" or similar

**Comment by jmcshane** [comment]
I realized the issue about the waypoint name as soon as I looked through the ambient integration tests. I think a name for the waypoint is better than maintaining the service account as an identifier

**Comment by ilrudie** [comment]
name sounds good to me. there is an issue for adding a naming option already... https://github.com/istio/istio/issues/49915

**Comment by jmcshane** [comment]
#49916 had merge conflicts, I brought the changes into this PR for now, can rebase later if desired.

Working on the integration tests now

File: istioctl/pkg/waypoint/waypoint.go [hunk 1]
```diff
@@ -373,7 +339,7 @@ func Cmd(ctx cli.Context) *cobra.Command {
 	waypointCmd.AddCommand(waypointGenerateCmd)
 	waypointCmd.AddCommand(waypointDeleteCmd)
 	waypointCmd.AddCommand(waypointListCmd)
-	waypointCmd.PersistentFlags().StringVarP(&waypointServiceAccount, "service-account", "s", "", "service account to create a waypoint for")
+	waypointCmd.PersistentFlags().StringVarP(&waypointName, "name", "", "default", "name of the waypoint")
 
 	_ = waypointCmd.RegisterFlagCompletionFunc("service-account", func(
```
Comment (MorrisLaw) [inline istioctl/pkg/waypoint/waypoint.go:344]: @howardjohn @ilrudie should I remove this in my PR as well?

Comment (MorrisLaw) [reply istioctl/pkg/waypoint/waypoint.go:344]: I'm guessing yes. It looks like this flag no longer exists, so we don't need flag completion.

**Pull request merged**